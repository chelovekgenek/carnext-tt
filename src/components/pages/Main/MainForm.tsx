import React from "react"
import { connect } from "react-redux"
import { FormikProps, Form, withFormik } from "formik"

import { Input, Checkbox } from "components/common/form"
import { Button } from "components/common/ui"

import { initialValues, schema } from "./schema"
import * as Styles from "./Main.styled"
import { TAppState } from "store/entities"
import { PromotionActions } from "store/entities/promotion"

interface IDispatchProps {
  submit: typeof PromotionActions.request
}
interface IProps extends IDispatchProps {}

export const MainForm: React.FC<IProps & FormikProps<typeof initialValues>> = ({ handleSubmit, isValid }) => (
  <Form>
    <Styles.Row>
      <Styles.Column padding={1}>
        <Input label="First name" name="firstname" />
      </Styles.Column>
      <Styles.Column padding={1}>
        <Input label="Surname" name="surname" />
      </Styles.Column>
    </Styles.Row>
    <Input label="Email address" name="email" />
    <Input label="Phone number" name="phone" />
    <Input label="Voucher code" name="code" />
    <Checkbox label="Accept Privacy Settlement" name="agreement" />
    <Button type="submit" onClick={handleSubmit} disabled={!isValid}>
      Get offer
      <i className="fas fa-chevron-right" />
    </Button>
  </Form>
)

export default connect<{}, IDispatchProps, {}, TAppState>(
  null,
  {
    submit: PromotionActions.request,
  },
)(
  withFormik<IProps, typeof initialValues>({
    handleSubmit: (values, { props }) => props.submit(values),
    mapPropsToValues: () => initialValues,
    validationSchema: schema,
    validateOnBlur: true,
  })(MainForm),
)
