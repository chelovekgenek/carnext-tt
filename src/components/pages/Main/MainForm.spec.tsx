import React from "react"
import { shallow } from "enzyme"

import formikMock from "mocks/formik-props"

import { MainForm } from "./MainForm"

describe("Main.MainForm", () => {
  const defaultValues = {
    firstname: "",
    surname: "",
    email: "",
    phone: "",
    code: "",
    agreement: true,
  }
  const render = (props = {}) =>
    shallow(<MainForm {...formikMock} initialValues={defaultValues} values={defaultValues} {...props} />)

  it("should render", () => {
    expect(render()).toMatchSnapshot()
  })
})
