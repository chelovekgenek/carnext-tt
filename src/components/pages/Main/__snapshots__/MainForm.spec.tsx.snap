// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main.MainForm should render 1`] = `
<Form>
  <styled.div>
    <styled.div
      padding={1}
    >
      <Input
        label="First name"
        name="firstname"
      />
    </styled.div>
    <styled.div
      padding={1}
    >
      <Input
        label="Surname"
        name="surname"
      />
    </styled.div>
  </styled.div>
  <Input
    label="Email address"
    name="email"
  />
  <Input
    label="Phone number"
    name="phone"
  />
  <Input
    label="Voucher code"
    name="code"
  />
  <Checkbox
    label="Accept Privacy Settlement"
    name="agreement"
  />
  <Button
    disabled={true}
    onClick={[MockFunction]}
    type="submit"
  >
    Get offer
    <i
      className="fas fa-chevron-right"
    />
  </Button>
</Form>
`;
